src/components/Reown/modals/SendTransactionModal.js

```diff
@@ -0,0 +1,198 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect } from 'react';
+import { t } from 'ttag';
+import { useSelector, useDispatch } from 'react-redux';
+import { constants, numberUtils } from '@hathor/wallet-lib';
+import { unregisteredTokensDownloadRequested } from '../../../actions';
+import { CopyButton } from '../../CopyButton';
+
+export function SendTransactionModal({ data, firstAddress, onAccept, onReject }) {
+  const dispatch = useDispatch();
+  const { tokenMetadata, tokens: registeredTokens } = useSelector((state) => ({
+    tokenMetadata: state.tokenMetadata,
+    tokens: state.tokens,
+  }));
+
+  useEffect(() => {
+    // Collect all unregistered token UIDs
+    const unregisteredTokens = new Set();
+    
+    data?.data?.inputs?.forEach(input => {
+      const token = registeredTokens.find(t => t.uid === input.token);
+      if (!token && input.token) {
+        unregisteredTokens.add(input.token);
+      }
+    });
+
+    data?.data?.outputs?.forEach(output => {
+      const token = registeredTokens.find(t => t.uid === output.token);
+      if (!token && output.token) {
+        unregisteredTokens.add(output.token);
+      }
+    });
+
+    if (unregisteredTokens.size > 0) {
+      dispatch(unregisteredTokensDownloadRequested(Array.from(unregisteredTokens)));
+    }
+  }, [data, registeredTokens, dispatch]);
+
+  const getTokenSymbol = (tokenId) => {
+    if (!tokenId) {
+      return constants.DEFAULT_NATIVE_TOKEN_CONFIG.symbol;
+    }
+
+    const token = registeredTokens.find(t => t.uid === tokenId);
+    if (token) {
+      return token.symbol;
+    }
+
+    return '?';
+  };
+
+  const formatValue = (value) => {
+    if (value == null) {
+      return '-';
+    }
+
+    return numberUtils.prettyValue(value);
+  };
+
+  const truncateTxId = (txId) => {
+    if (!txId) return '-';
+    return `${txId.slice(0, 8)}...${txId.slice(-8)}`;
+  };
+
+  const renderInputs = () => {
+    if (!data?.data?.inputs || data.data.inputs.length === 0) {
+      return null;
+    }
+
+    return (
+      <div className="mb-4">
+        <h6 className="mb-3">{t`Inputs`}</h6>
+        {data.data.inputs.map((input, index) => (
+          <div key={index} className="p-3 bg-light rounded mb-2">
+            <div className="d-flex justify-content-between align-items-start mb-2">
+              <div>
+                <strong>{t`Input ${index + 1}`}</strong>
+              </div>
+              <div>
+                {formatValue(input?.value)} {getTokenSymbol(input?.token)}
+              </div>
+            </div>
+            <div className="text-monospace">
+              {truncateTxId(input?.txId)} ({input?.index})
+              {input?.txId && (
+                <button 
+                  className="btn btn-link btn-sm p-0 ml-2" 
+                  onClick={() => navigator.clipboard.writeText(input.txId)}
+                >
+                  <i className="fa fa-copy"></i>
+                </button>
```

Comment:
Why don't you use the `CopyButton` component?

Reply:
Refactored to use it, thanks